src/Aspire.Hosting.NodeJs/NodeExtensions.cs

```diff
@@ -187,18 +187,19 @@ public static IResourceBuilder<ViteAppResource> AddViteApp(this IDistributedAppl
     }
 
     /// <summary>
-    /// Ensures the Node.js packages are installed before the application starts using npm as the package manager.
+    /// Configures the Node.js resource to use npm as the package manager and optionally installs packages before the application starts.
     /// </summary>
     /// <param name="resource">The NodeAppResource.</param>
+    /// <param name="autoInstall">When true (default), automatically installs packages before the application starts. When false, only sets the package manager annotation without creating an installer resource.</param>
     /// <param name="useCI">When true, use <code>npm ci</code>, otherwise use <code>npm install</code> when installing packages.</param>
-    /// <param name="configureInstaller">Configure the npm installer resource.</param>
+    /// <param name="configureInstaller">Configure the npm installer resource. Only applicable when <paramref name="autoInstall"/> is true.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<TResource> WithNpmPackageManager<TResource>(this IResourceBuilder<TResource> resource, bool useCI = false, Action<IResourceBuilder<NodeInstallerResource>>? configureInstaller = null) where TResource : NodeAppResource
+    public static IResourceBuilder<TResource> WithNpm<TResource>(this IResourceBuilder<TResource> resource, bool autoInstall = true, bool useCI = false, Action<IResourceBuilder<NodeInstallerResource>>? configureInstaller = null) where TResource : NodeAppResource
```

Comment:
```suggestion
    public static IResourceBuilder<TResource> WithNpm<TResource>(this IResourceBuilder<TResource> resource, bool install = false, bool useCI = false, Action<IResourceBuilder<NodeInstallerResource>>? configureInstaller = null) where TResource : NodeAppResource
```

Change the autoInstall to be `false` by default and rename it to `install`.

With this change, the apphost.cs file in the starter template needs to be updated to `.WithNpm(install: true)`